src/wizards/tokenSwapDealWizard/wizardManager.ts

```diff
@@ -188,11 +188,21 @@ export class WizardManager implements IRouteableComponent {
 
   public async next() {
     const result = await this.controller.validate();
+    this.stages[this.activeIndex].valid = result.valid;
     if (!result.valid) {
       this.eventAggregator.publish("handleValidationError", "Unable to proceed, please check the page for validation errors");
       return;
     }
 
+    // Make sure all the stages are valid before continuing
+    if (this.wizardType === WizardType.createOpenProposal || this.wizardType === WizardType.createPartneredDeal) {
```

Comment:
@IonelLupu 
Do you perhaps know why this has to be added in Au2, but was not needed in Au1?

Reply:
it was in AU1 [here](https://github.com/PrimeDAO/prime-deals-dapp/blob/development/src/wizards/services/WizardService.ts#L138)

The wizard was built a bit different but the functionality was almost the same